test/common/upstream/cluster_manager_impl_test.cc

```diff
@@ -260,6 +260,50 @@ TEST_F(ClusterManagerImplTest, UnknownHcType) {
   EXPECT_THROW(create(*loader), EnvoyException);
 }
 
+TEST_F(ClusterManagerImplTest, MaxClusterName) {
+  std::string json = R"EOF(
+  {
+    "clusters": [
+    {
+      "name": "clusterwithareallyreallylongnamemorethanmaxcharsallowedbyschema"
+    }]
+  }
+  )EOF";
+
+  Json::ObjectPtr loader = Json::Factory::LoadFromString(json);
+  try {
+    create(*loader);
+    ADD_FAILURE() << "Json::Exception should take place. It did not.";
```

Comment:
This is a common pattern that we will do again (and should do more). make a macro like EXPECT_THROW(), EXPECT_THROW_WITH_MESSAGE(), put it somewhere common, then use it.